### ISSUE TITLE
Issue with RegEx replacing in JSON file

### ISSUE THREAD

**Issue by omarisai** [microsoft/vscode#172041]
Type: <b>Bug</b>

Description:
There is an issue when replacing text using RegEx expressions and the files where the text is, are closed.

Steps to reproduce:

1. Open the provided folder with VS code.
2. close the 'Get Started' page and don't open any files.
3. Open the global 'Search' tab (Ctrl+Shift+H).
4. Enable only RegEx (the .* button) and search for: \[\n\s+([\d]+),\n\s+([\d]+)\n\s+\]
5. Fill the Replace box with: [ $1, $2 ]
![image](https://user-images.githubusercontent.com/10107642/214117471-a1a74050-1f0f-4611-baf2-77bf130d80b7.png)
7. Click on the second search/replace result.
![image](https://user-images.githubusercontent.com/10107642/214117552-f51e5633-fd33-45cc-a22e-8003cbea2ac7.png)

Observed behavior:

1. VS code opens on the 'Get Started' page.
2. The 'Get Started' page is closed.
3. The 'Search' is opened.
4. The search results are shown.
5. The replace preview is shown in the 'Search' tab, the 2nd replacement is [ 1, 2 ], which is wrong.
7. VS code opens a file replace preview, 2nd replacement is still incorrect there.

Expected behavior:

5. The replace preview should show [ 3, 4 ].
6. The file replace preview should show [ 3, 4 ].

VS Code version: Code 1.74.3 (97dec172d3256f8ca4bfb2143f3f76b503ca0534, 2023-01-09T16:59:02.252Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz (8 x 3392)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.96GB (9.63GB free)|
|Process Argv|C:\\Users\\omari\\Documents\\minecraft\\estable\\BasicStuff.code-workspace --crash-reporter-id bdf29225-8222-4fd0-a5ed-a866da8f6f62|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
blockceptionvscodeminecraftbedrockdevelopmentextension|Blo|5.0.5
doxdocgen|csc|1.4.0
vscode-office|cwe|2.9.5
gitlab-workflow|Git|3.59.2
pddl|jan|2.25.1
jbockle-format-files|jbo|3.4.0
plantuml|jeb|2.17.5
better-cpp-syntax|jef|1.17.2
csharp|ms-|1.25.2
isort|ms-|2022.8.0
python|ms-|2022.20.2
vscode-pylance|ms-|2023.1.30
jupyter|ms-|2022.11.1003412109
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.14
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.266.1
remote-ssh|ms-|0.94.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.72.0
cmake-tools|ms-|1.13.40
cpptools|ms-|1.13.9
cpptools-extension-pack|ms-|1.3.0
powershell|ms-|2023.1.0
remote-explorer|ms-|0.0.3
cmake|twx|0.0.17
luna-paint|Tyr|0.16.0
vscode-icons|vsc|12.2.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
pyindex848:30577860
nodejswelcome1:30587005
282f8724:30602487
89544117:30613380
vscup:30628656

```

</details>

**Comment by ArturoDent** [comment]
Can you try this in the Insiders Build.  I was able to replicate your issue in the v1.74.3 Stable Build, but **not** in the v1.75 Insiders Build, so it may be that this has been fixed in the last two months.

**Comment by omarisai** [comment]
Sure, I will try.

**Comment by omarisai** [comment]
BTW, thanks for the fast response 👍

**Comment by omarisai** [comment]
@ArturoDent, you are right, I tested with the insiders build and I cannot reproduce the issue.

**Comment by ArturoDent** [comment]
I recall some work being done on similar issues recently, but we would have to wait for an official response.  Thanks for testing on Insiders.

**Comment by roblourens** [comment]
Thanks for checking 1.75
